### PR TITLE
feat(provider/aws): Add accountId to subnet responses

### PR DESCRIPTION
**Pull request by gardleopard**
accountId added to the subnet object for aws and ecs.

```
{
  "account" : "default",
  "accountId" : "402251857394",
  "availabilityZone" : "eu-west-1c",
  "availableIpAddressCount" : 4091,
  "cidrBlock" : "172.31.16.0/20",
  "deprecated" : false,
  "id" : "subnet-0432b02b",
  "purpose" : null,
  "region" : "eu-west-1",
  "state" : "available",
  "target" : null,
  "type" : "aws",
  "vpcId" : "vpc-d023c5b4"
}
```

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonSubnetProvider.groovy [hunk 1]
```diff
@@ -96,6 +106,7 @@ class AmazonSubnetProvider implements SubnetProvider<AmazonSubnet> {
         purpose = "${splits[1]} (${splits[0]})"
       }
     }
+    def accountId = accounts.find{ account -> account.name.equals(parts.account)}.accountId
```
Comment (gardleopard) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonSubnetProvider.groovy]: Can we get a miss here? If one account is removed and the clouddriver cache has stale data here?

Comment (cfieber) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonSubnetProvider.groovy]: yeah actually that could happen

one thought would be to rework this and stick the accountId into the cached data when it is stored? (would eliminate the need to resolve it on the way back out)

Comment (gardleopard) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonSubnetProvider.groovy]: Reworked in https://github.com/spinnaker/clouddriver/pull/3101

**Comment by gardleopard** [comment]
Closing this because https://github.com/spinnaker/clouddriver/pull/3101 is probably a better way to solve this.

**Pull request closed without merging**